app/src/main/java/a3/com/convo/adapters/FriendAdapter.java

```diff
@@ -53,61 +66,35 @@ public FriendAdapter(ArrayList<String> friends) {
     @Override
     public void onBindViewHolder(@NonNull final FriendAdapter.ViewHolder holder, int position) {
         holder.itemView.setBackgroundColor(selectedPos == position ? Color.rgb(229,229,229) : Color.TRANSPARENT);
-        // Get friends user ID's
-        final String friendId = myFriends.get(position);
-        final String profilePicPath = "/" + friendId + "/picture";
-
-        //Get friends profile picture
-//        GraphRequest picRequest = GraphRequest.newGraphPathRequest(AccessToken.getCurrentAccessToken(), "/" + friendId + "/picture", new GraphRequest.Callback() {
-//            @Override
-//            public void onCompleted(GraphResponse response) {
-//                try {
-//
-//                } catch (JSONException e) {
-//
-//                }
-//            }
-//        });
-
-        GraphRequest picRequest = GraphRequest.newMeRequest(
-                AccessToken.getCurrentAccessToken(),
-                new GraphRequest.GraphJSONObjectCallback() {
-                    @Override
-                    public void onCompleted(
-                            JSONObject json_object,
-                            GraphResponse response) {
-                        try {
-                            JSONObject picInfo = json_object.getJSONObject("picture").getJSONObject("data");
-                            String picURL = picInfo.optString("url");
-                            GlideApp.with(context)
-                                    .load(picURL)
-                                    .circleCrop()
-                                    .into(holder.ivFriend);
-                        } catch(Exception e){
-                            e.printStackTrace();
-                            Toast.makeText(context, "Pic URL not retrieved", Toast.LENGTH_SHORT).show();
-                        }
+        // Get friend
+        final String friend = myFriends.get(position);
+        // Find friend in background
+        ParseQuery<ParseUser> query = ParseUser.getQuery();
+        query.whereEqualTo("username", friend);
+        query.findInBackground(new FindCallback<ParseUser>() {
+            @Override
+            public void done(List<ParseUser> objects, ParseException e) {
+                if (e == null) {
+                    // Get object's values from parse
+                    currentFriend = objects.get(0);
+                    profPic = currentFriend.getString("profPicUrl");
+                    name = currentFriend.getString("name");
+
+                    // Set friend's photo
+                    if(profPic != null) {
+                        GlideApp.with(context)
```

Comment:
should we check if context is null? Verify if GlideApp.with() takes null values. If not, you can add a check here.